### PR TITLE
Fix base types of second paramters used in Avx.PermuteVar and Avx2.Permute8x32 for float, double overloads

### LINKED ISSUE #19579: Avx2.Permute8x32 and Avx.PermuteVar - wrong mask base type for Single and Double overloads
Definition of Avx2.Permute8x32 has wrong type of second parameter:
```C#
        /// <summary>
        /// __m256 _mm256_permutevar8x32_ps (__m256 a, __m256i idx)
        ///   VPERMPS ymm, ymm/m256, imm8
        /// </summary>
        public static Vector256<float> PermuteVar8x32(Vector256<float> left, Vector256<float> mask) => PermuteVar8x32(left, mask);
```
Intel intrinsics single overload has the following signature:

```C
    __m256 _mm256_permutevar8x32_ps (__m256 a, __m256i idx)
```

The reason for second parameter (third instruction operand) being integral type is that it is used to express index values used during permutation where 3 first bits of each integer are used to encode destination location. Trying to express that pattern using single floats could be mind boggling.

### PR DESCRIPTION
**Pull request by 4creators**
Fixes #19579 

cc @CarolEidt @eerhardt @fiigii @tannergooding

### CODE REVIEW

**Comment by CarolEidt** [comment]
The changes look good, but it seems that the tests will need to be updated as well.

**Comment by 4creators** [comment]
> The changes look good, but it seems that the tests will need to be updated as well.

Working on this

File: src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx.PlatformNotSupported.cs [hunk 1]
```diff
@@ -761,22 +761,22 @@ public abstract class Avx : Sse42
         /// __m128 _mm_permutevar_ps (__m128 a, __m128i b)
         ///   VPERMILPS xmm, xmm, xmm/m128
         /// </summary>
-        public static Vector128<float> PermuteVar(Vector128<float> left, Vector128<float> mask) { throw new PlatformNotSupportedException(); }
+        public static Vector128<float> PermuteVar(Vector128<float> left, Vector128<int> mask) { throw new PlatformNotSupportedException(); }
```
Comment (fiigii) [inline src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx.PlatformNotSupported.cs]: Please change the parameter name to `control` that is not a "mask".

**Comment by fiigii** [comment]
>  but it seems that the tests will need to be updated as well.

You probably need to disable (remove) the current tests and update them after CoreFX change refect to CoreCLR repo.

File: src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx2.cs [hunk 2]
```diff
@@ -1294,7 +1294,7 @@ public abstract class Avx2 : Avx
         /// __m256 _mm256_permutevar8x32_ps (__m256 a, __m256i idx)
         ///   VPERMPS ymm, ymm/m256, imm8
         /// </summary>
-        public static Vector256<float> PermuteVar8x32(Vector256<float> left, Vector256<float> mask) => PermuteVar8x32(left, mask);
+        public static Vector256<float> PermuteVar8x32(Vector256<float> left, Vector256<int> mask) => PermuteVar8x32(left, mask);
```
Comment (fiigii) [inline src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx2.cs]: Please change the parameter name to control that is not a "mask".

File: src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx.PlatformNotSupported.cs (hunk 1 above)
Comment (tannergooding) [inline src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx.PlatformNotSupported.cs]: Why? I thought that, elsewhere, we had explicitly kept these the same type (and we have had several PRs moving these off of the integer types and onto the same type as the first parameter)....

This definitely needs to be discussed during the next design-review session. CC. @eerhardt, @terrajobst

Comment (fiigii) [reply src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx.PlatformNotSupported.cs]: We only keep the same base type for mask vectors, but here is actually not mask. That is why I suggest to rename to “control”.

Comment (fiigii) [reply src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx.PlatformNotSupported.cs]: You can see some intrinsic that takes “index” vector always with integer base type.

Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx.PlatformNotSupported.cs]: Ah, I see, 👍 then.

**Comment by 4creators** [comment]
Updated PR, addressed code review feedback and removed conflicting tests. Adding tests after related changes in corefx repo are absorbed is tracked by #19624

**Comment by tannergooding** [comment]
@4creators, could you get the corresponding CoreFX PR up and ready. Once that is done, we should be good to merge.

**Comment by 4creators** [comment]
@tannergooding Done PTAL

**Comment by tannergooding** [comment]
Thanks for the work!

**Pull request merged**